src/main/java/org/cbioportal/web/columnar/BasicDataBinner.java

```diff
@@ -0,0 +1,306 @@
+                filteredClinicalDataCounts = studyViewColumnarService.getGenericAssayDataBinCounts(studyViewFilter, uniqueKeys);
+                attributeDatatypeMap = Collections.emptyMap();
+            }
+            default -> {
+                unfilteredClinicalDataCounts = Collections.emptyList();
+                filteredClinicalDataCounts = Collections.emptyList();
+                attributeDatatypeMap = Collections.emptyMap();
+            }
+        }
+
+        // TODO ignoring conflictingPatientAttributeIds for now
+        List<ClinicalData> unfilteredClinicalData = convertCountsToData(
+            unfilteredClinicalDataCounts.stream().flatMap(c -> c.getCounts().stream()).toList()
+        );
+        List<ClinicalData> filteredClinicalData = convertCountsToData(
+            filteredClinicalDataCounts.stream().flatMap(c -> c.getCounts().stream()).toList()
+        );
+        
+        Map<String, List<Binnable>> unfilteredClinicalDataByAttributeId =
+            unfilteredClinicalData.stream().collect(Collectors.groupingBy(Binnable::getAttrId));
+
+        Map<String, List<Binnable>> filteredClinicalDataByAttributeId =
+            filteredClinicalData.stream().collect(Collectors.groupingBy(Binnable::getAttrId));
+
+        // Define result variables
+        List<U> resultDataBins = Collections.emptyList();
+
+        // TODO: need to update attributeDatatypeMap to include patient level data for Generic Assay Profiles
+        if (dataBinMethod == DataBinMethod.STATIC) {
+            if (!unfilteredClinicalData.isEmpty()) {
+                resultDataBins = calculateStaticDataBins(
+                    dataBinner,
+                    dataBinFilters,
+                    attributeDatatypeMap,
+                    unfilteredClinicalDataByAttributeId,
+                    filteredClinicalDataByAttributeId
+                );
+            }
+        }
+        // TODO: need to update attributeDatatypeMap to include patient level data for Generic Assay Profiles
+        else { // dataBinMethod == DataBinMethod.DYNAMIC
+            // TODO we should consider removing dynamic binning support
+            //  we never use dynamic binning in the frontend because number of bins and the bin ranges can change 
+            //  each time there is a new filter which makes the frontend implementation complicated
+            if (!filteredClinicalData.isEmpty()) {
+                resultDataBins = calculateDynamicDataBins(
+                    dataBinner,
+                    dataBinFilters,
+                    attributeDatatypeMap,
+                    filteredClinicalDataByAttributeId
+                );
+            }
+        }
+
+        return resultDataBins;
+    }
+
+    private <S extends DataBinFilter> void removeSelfFromFilter(S dataBinFilter, StudyViewFilter studyViewFilter) {
+        if (studyViewFilter != null) {
```

Comment:
I would simply this and force studyViewFilter not to be null.

Then you can do what you did above and do switch with pattern matching and remove the need for instanceof